### PR TITLE
BLD: build fridge

### PR DESCRIPTION
**Pull request by eigenfoo**

### CODE REVIEW

File: src/fridge.py [hunk 1]
```diff
@@ -1,42 +1,69 @@
...
+        'optimizer_disc': optimizer_disc,
+    }
+    state = {
+        'epoch': epoch,
+        'gen_state_dict': gen.state_dict(),
+        'disc_state_dict': disc.state_dict(),
+        'optimizer_gen_state_dict': optimizer_gen.state_dict(),
+        'optimizer_disc_state_dict': optimizer_disc.state_dict(),
+    }
+    torch.save(model, 'checkpoints/models/model_{}_{}.pt'.format(
```
Comment (ccurro) [inline src/fridge.py]: I think we're using python3.6? If so please use f-strings. 

https://www.python.org/dev/peps/pep-0498/

File: src/fridge.py [hunk 2]
```diff
@@ -1,42 +1,69 @@
...
+    if path[:6] == 'model_':
+        checkpoint = torch.load(path)
+        epoch = checkpoint['epoch']
+        gen = checkpoint['gen']
+        disc = checkpoint['disc']
+        optimizer_gen = checkpoint['optimizer_gen']
+        optimizer_disc = checkpoint['optimizer_disc']
+        return epoch, gen, disc, optimizer_gen, optimizer_disc
+    elif path[:6] == 'state_':
+        msg = 'Must instatiate correct class. See https://pytorch.org/docs/stable/notes/serialization.html for more details.'
```
Comment (ccurro) [inline src/fridge.py]: If we can't load state, why are we saving it? Is there unwritten load code?

File: src/infer.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's',
+    't', 'u', 'v', 'w', 'x', 'y', 'z', 'zero', 'one', 'two', 'three', 'four',
+    'five', 'six', 'seven', 'eight', 'nine', 'exclam', 'numbersign', 'dollar',
+    'percent', 'ampersand', 'asterisk', 'question', 'at'
+]
+
+DEVICE = torch.device("cuda:1" if torch.cuda.is_available() else "cpu")
+
+
+def rasterize(x, resolution=64, sigma=0.01):
```
Comment (ccurro) [inline src/infer.py]: Is there a reason this isn't' being imported from elsewhere?

File: src/infer.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    y_meshgrid_expanded = y_meshgrid.expand(batch_size, resolution, resolution,
+                                            num_samples)
+
+    raster = torch.exp((-(x_samples - x_meshgrid_expanded)**2 -
+                        (y_samples - y_meshgrid_expanded)**2) / (2 * sigma**2))
+    raster = raster.sum(dim=3)
+    return raster
+
+
+def save_images(images, size, path):
```
Comment (ccurro) [inline src/infer.py]: best put in this in a util.py type of thing b/c the periodic val in the train script will use it too

File: src/fridge.py (hunk 1 above)
Comment (eigenfoo) [reply src/fridge.py]: We're only py3.5 on kahan.

File: src/infer.py (hunk 3 above)
Comment (eigenfoo) [reply src/infer.py]: I'd have to import a class to do it. tbh it's probably worth moving out into a `utils.py`, along with `CHARACTERS` and other things

File: src/fridge.py (hunk 2 above)
Comment (eigenfoo) [reply src/fridge.py]: Loading state requires you to know which pastry it is, and which optimizers you used. Without this knowledge, I can't load it in.

**Comment by eigenfoo** [comment]
For @ArianaFreitag @ostapstephan:

https://gist.github.com/ccurro/0eb4851d898ec8c8397c2796e011b593 may be helpful for the fridge.

**Comment by eigenfoo** [comment]
Superseded by #78. Starting a fresh branch with fresh eyes.

**Pull request closed without merging**